distributed/deploy/spec.py

```diff
@@ -473,10 +473,13 @@ def new_worker_spec(self):
         --------
         scale
         """
-        while self._i in self.worker_spec:
+        worker_name_template = f"{self._name}-{{}}"
+        worker_name = worker_name_template.format(self._i)
```

Comment:
What if instead we generalize this to having a `SpecCluster._new_worker_name` method or something similar?  Today this would be the following by default

```python
def _new_worker_name(self):
    while self._i in self.worker_spec:
        self._i += 1
    return self._i
```

But Dask-JobQueue could overwrite it on its own method.

This is just one idea, but shows how we might generailize something on SpecCluster that would make it easier to modify downstream without changing default behaviors on SpecCluster itself.

Reply:
This seems completely reasonable (I thought about adding a `worker_name_func` as an argument to `SpecCluster.__init__` which seems slightly overcomplicated but not the good old object-oriented approach), I'll do this and add some tests as well.

Reply:
I added the `_new_worker_name(self, worker_number)` method. This feels simpler to customise in Dask-Jobqueue (no need for the `self._i` updating logic).

I also added some simple tests.